src/Symfony/Component/Ldap/LdapClient.php

```diff
@@ -41,7 +43,7 @@ class LdapClient implements LdapClientInterface
      * @param bool   $useStartTls
      * @param bool   $optReferrals
      */
-    public function __construct($host = null, $port = 389, $version = 3, $useSsl = false, $useStartTls = false, $optReferrals = false)
+    public function __construct($host = null, $port = 389, $version = 3, $useSsl = false, $useStartTls = false, $optReferrals = false, $ldapBaseDn = null, $ldapSearchDn = null, $ldapSearchPassword = null, $ldapUidKey = null, $ldapFilter = null)
```

Comment:
@csarrazi : these fields are passed to the LdapUserProvider inside the LdapBindAuthenticationProvider. This is required in the cases where another than the LdapUserProvider is used for all other purposes than authentication (as described above). Refer to line 87 in LdapBindAuthenticationProvider.php